src/lib/fragments/AgGrid.react.js

```diff
@@ -192,15 +194,62 @@ export default class DashAgGrid extends Component {
 
     setSelection(selection) {
         const {gridApi} = this.state;
+        const {getRowId} = this.props;
         if (gridApi && selection) {
-            if (!selection.length) {
+            this.setState({pauseSelections: true});
+            if (Object.keys(selection).includes('function')) {
+                // keeps grid from rendering display unnecessarily
+                gridApi.deselectAll();
+                const parsedCondition = esprima.parse(selection.function)
+                    .body[0].expression;
+                const context = {
+                    d3,
+                    ...customFunctions,
+                    ...window.dashAgGridFunctions,
+                };
+                const test = function (node) {
+                    if (evaluate(parsedCondition, {node, ...context})) {
+                        node.setSelected(true);
+                    }
+                };
+                gridApi.forEachNode(test);
+            } else if (Object.keys(selection).includes('ids')) {
+                // keeps grid from rendering display unnecessarily
                 gridApi.deselectAll();
-            } else {
                 gridApi.forEachNode((node) => {
-                    const isSelected = selection.some(equals(node.data));
-                    node.setSelected(isSelected);
+                    if (selection.ids[node.id]) {
+                        node.setSelected(true);
+                    }
                 });
+            } else {
+                if (!selection.length) {
+                    gridApi.deselectAll();
+                } else {
+                    // keeps grid from rendering display unnecessarily
+                    gridApi.deselectAll();
+                    if (getRowId) {
+                        const parsedCondition = esprima.parse(
+                            getRowId.replaceAll('params.data.', '')
+                        ).body[0].expression;
+                        const mapId = {};
+                        selection.map((params) => {
```

Comment:
```suggestion
                        selection.forEach((params) => {
```
`map` works, but since we aren't using the return value it's best to use `forEach`.